diff --git a/node_modules/react-native-gesture-handler/android/build.gradle b/node_modules/react-native-gesture-handler/android/build.gradle
index ac149a0..7a31d09 100644
--- a/node_modules/react-native-gesture-handler/android/build.gradle
+++ b/node_modules/react-native-gesture-handler/android/build.gradle
@@ -236,7 +236,7 @@ dependencies {
 
     if (shouldUseCommonInterfaceFromReanimated()) {
         // Include Reanimated as dependency to load the common interface
-        implementation(rootProject.subprojects.find { it.name == 'react-native-reanimated' }) {
+        compileOnly(rootProject.subprojects.find { it.name == 'react-native-reanimated' }) {
             // resolves "Duplicate class com.facebook.jni.CppException"
             exclude group: 'com.facebook.fbjni'
         }
